plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/LaunchAppEngineStandardHandler.java

```diff
@@ -0,0 +1,132 @@
+
+package com.google.cloud.tools.eclipse.appengine.localserver.ui;
+
+import com.google.cloud.tools.eclipse.appengine.localserver.server.LocalAppEngineServerDelegate;
+import com.google.cloud.tools.eclipse.util.AdapterUtil;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IFileEditorInput;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.wst.server.core.IModule;
+import org.eclipse.wst.server.core.IServer;
+import org.eclipse.wst.server.core.IServerType;
+import org.eclipse.wst.server.core.IServerWorkingCopy;
+import org.eclipse.wst.server.core.ServerCore;
+import org.eclipse.wst.server.core.ServerUtil;
+
+public class LaunchAppEngineStandardHandler extends AbstractHandler {
+  @VisibleForTesting
+  boolean mockLaunch = false;
+
+  @Override
+  public Object execute(ExecutionEvent event) throws ExecutionException {
+    String launchMode = event.getParameter("launchMode");
+    if (launchMode == null) {
+      launchMode = ILaunchManager.DEBUG_MODE;
+    }
+    IModule[] modules = asModules(event);
+    SubMonitor progress = SubMonitor.convert(null, 10);
+    try {
+      IServer server = findExistingServer(modules, progress.newChild(3));
```

Comment:
Yeah, that's what I'm talking. I may be missing something, but if we are in the case I described above, shouldn't we either stop the existing server and restart it in the right mode, or create a new server in the right mode?